src/metabase/api/card.clj

```diff
@@ -194,6 +195,7 @@
                     :can_manage_db
                     [:collection :is_personal]
                     [:moderation_reviews :moderator_details])
+        (update :dashboard #(when % (select-keys % [:name :id :moderation_status])))
```

Comment:
The case I'm trying to avoid is `{:dashboard nil}` turning into `{:dashboard {}}`, so the key is actually present just set to `nil`.

Maybe instead I can just do `(update :dashboard #(some-> % (select-keys [:name :id :moderation_status])))`? A little less ugly than what I have right now?